### ISSUE TITLE
Returning Different Status Codes

### ISSUE THREAD

**Issue by aklatt1194** [toumorokoshi/transmute-core#4]
Not sure if this is a bug or a feature request...

It could just be me being dumb and missing it, but it seems like there is not a way to return status codes other than 200 without raising an APIException (which comes along with its own issues as that forces your response to have the generic error message)? (At least that is my interpretation from looking at: https://github.com/toumorokoshi/transmute-core/blob/master/transmute_core/handler.py)

I think it would be nice to have a little bit more control over what status codes and response body you return.

Thanks!
-AJ

**Comment by toumorokoshi** [comment]
@aklatt1194 agreed it would be nice be flexible about that. It's pretty trivial for transmute-core to provide some sort of response object to allow that customization:

```
import transmute_core

def foo():
    return transmute_core.response({}, status=201)
```

And then handles that appropriately in the method. However, the tricky aspect comes in on the autodocumentation.

We need to document the behavioral change in the swagger spec. There's an object of <status_code, response_type> pairs that is dictated in an API spec:

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#paths-object-example

It would be great for the API to be documented there as well.

To do that, I could imagine another annotation in describe:

```
@describe(..., status_code=201)
def foo():
    return {}
```

But I'm wondering if there's some way to document multiple different response types, of different schemas, without requiring explicit declarations all over the place:

```
@describe(response_type_by_code={200: ModelA(), 201: ModelB()})
```

Thoughts?

**Comment by aklatt1194** [comment]
Hmm... 
Ya, I agree that it would be good for the auto-documentation to have a way to pick that up. I think that your proposal makes sense to me (specifying the possible response codes and their associated models as a map in the describe annotation). That might be the cleanest option...

**Comment by toumorokoshi** [comment]
fixed in v0.5.0 @aklatt1194 I've gone a slightly different direction with the format. note the docs:

http://transmute-core.readthedocs.io/en/latest/function.html#multiple-response-types
